### PR TITLE
Adding support for functions from rms packages

### PR DESCRIPTION
**Pull request by RomanTsegelskyi**
This PR is to add support for markdown rendering for functions from `rms`. I had to close and open new one because I messed up my branches a bit.

### CODE REVIEW

**Comment by RomanTsegelskyi** [comment]
To complete `pander.lrm`, we need to wait https://github.com/harrelfe/rms/issues/12 is resolved. I encountered it while trying to figure out how `lrm` works

File: R/S3.R [hunk 1]
```diff
@@ -1805,3 +1805,190 @@ pander.nls <- function(x, digits = panderOptions('digits'), show.convergence = F
     pander.summary.nls(summary(x), summary = FALSE, add.significance.stars = FALSE,
                        digits = digits, show.convergence = show.convergence, ...)
 }
+
+multitable <- function(...) {
```
Comment (daroczig) [inline R/S3.R]: Can you move this to `helpers.R` with some documentation?

**Comment by daroczig** [comment]
Thank you very much again for the fix, @harrelfe -- and sorry for posting it again, I've just seen your comment here, I should have originally answered here on the approach: although `pandoc` and `markdown` are indeed quite limited in what the syntax offers for tables, but this limited syntax supports seamless conversion not only to LaTeX or HTML, but to docx, odt and a bunch of other document formats as well -- which is a huge advantage in some use cases.

Also, if working with `Pandoc.brew`, you do not have to think about if calling `xtable`, `latex`, `html`, `stargazer`, `broom`, `kable`, `print` or any other function is required -- all R objects are automatically transformed to the appropriate markdown format, so that the author can concentrate on writing the article and R code, not trying to figure out how to represent an R object in LaTeX, HTML etc.

**Comment by daroczig** [comment]
Impressive job, @RomanTsegelskyi, thank you!

@Hjalt, if you have some time, check out this branch and have a look at the current output of the S3 methods:

```r
devtools::install_github('RomanTsegelskyi/pander@rms')
```

**Comment by RomanTsegelskyi** [comment]
It seems like there is an issue with `roxygen2` - https://github.com/klutometis/roxygen/issues/368 that it doesn't handle `pander.glm` and `pander.Glm` correctly. Any ideas for WA?

**Comment by daroczig** [comment]
Seems to run fine here:

```
[daroczig@localhost][~/projects/pander]% R --vanilla                                                                                                                                                 ⏎

R version 3.2.0 (2015-04-16) -- "Full of Ingredients"
Copyright (C) 2015 The R Foundation for Statistical Computing
Platform: x86_64-unknown-linux-gnu (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> library(roxygen2)
> roxygenize()
Loading required package: digest
Loading required package: tools
Loading required package: Rcpp
Writing pander.glm.Rd
Writing pander.Glm.Rd
> list.files(path = 'man', pattern = '[gG]lm')
[1] "pander.Glm.Rd"         "pander.glm.Rd"         "pander.summary.glm.Rd"
> sessionInfo()
R version 3.2.0 (2015-04-16)
Platform: x86_64-unknown-linux-gnu (64-bit)
Running under: Arch Linux

locale:
 [1] LC_CTYPE=hu_HU.utf8       LC_NUMERIC=C             
 [3] LC_TIME=hu_HU.utf8        LC_COLLATE=C             
 [5] LC_MONETARY=hu_HU.utf8    LC_MESSAGES=hu_HU.utf8   
 [7] LC_PAPER=hu_HU.utf8       LC_NAME=C                
 [9] LC_ADDRESS=C              LC_TELEPHONE=C           
[11] LC_MEASUREMENT=hu_HU.utf8 LC_IDENTIFICATION=C      

attached base packages:
[1] tools     stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] Rcpp_0.11.6    digest_0.6.8   roxygen2_4.1.1

loaded via a namespace (and not attached):
[1] magrittr_1.5     codetools_0.2-11 stringi_0.5-5    stringr_1.0.0   
```

You are not on Windows, right?

**Comment by RomanTsegelskyi** [comment]
That's interesting. I have a make. And same code doesn't generate it

```r
> roxygenize()
Writing NAMESPACE
Writing pander.glm.Rd
Writing pander.Glm.Rd
Deleting test.Rd
> list.files(path = 'man', pattern = '[gG]lm')
[1] "pander.glm.Rd"         "pander.summary.glm.Rd"
> sessionInfo()
R version 3.2.1 (2015-06-18)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: OS X 10.10.4 (Yosemite)

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] tools     grid      stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] roxygen2_4.1.1       pander_0.5.3         gtable_0.1.2         tseries_0.10-34     
```

**Comment by daroczig** [comment]
But your output says:

```
Writing pander.glm.Rd
Writing pander.Glm.Rd
```

What's the file system?

**Comment by daroczig** [comment]
@RomanTsegelskyi see e.g. http://apple.stackexchange.com/a/71360
But I'm happy to run `roxygen` after merging to keep it simple.

**Comment by Hjalt** [comment]
I'm amazed at your efforts so far: fantastic job! I'm "unfortunately" on vacation right now, but i'll check it put as soon as I get back.

**Comment by RomanTsegelskyi** [comment]
@daroczig thanks for the link. Turns out both my Windows and Mac are non-case sensitive. Thanks for offering to run it on your machine, but I decided to fix it by specifying `@rdname pander.glm.rms` to avoid possible overrides when doing `roxygenize` in future.

**Comment by daroczig** [comment]
You should rather get a proper filesystem :)

**Comment by harrelfe** [comment]
For the reasons I've given before, this is a high-maintenance way to go about it.  Future changes in rms will break this approach, and you are not achieving the tabular and math capabilities with print(fit object, latex=TRUE).  There are two more general approaches: Install the system packages TeX4ht or latexml and run either htlatex or latexml on the latex code produced by print(..., latex=TRUE).  Then you'll get greek letters, graphics for missing data frequencies, and nice tabular output for model stats and for coefficients.  latexml can produce html5 using mathml.

**Comment by daroczig** [comment]
Thank you very much for your feedback, @harrelfe. Unfortunately, we cannot use TeX or HTML-only methods, as markdown conversion supports many other formats as well.

For @Hjalt's `fit1` [example](https://github.com/Rapporter/pander/issues/207#issuecomment-124820622) we get the following LaTeX/pdf output:

latex-outut

I don't see anything that is not supported by Pandoc's markdown -- which supports TeX math formulas between the dollar signs with all features (even after converting markdown to e.g. MS Word). Based on this, technically it's possible to add support for the math markup (using the very same LaTeX formulas as found in`rms:::print.ols` -- although this would make the markdown implementation even more complex with a not too DRY approach.

Currently, that's what we generate:

```
> pander::pander(fit1)

---------------------------------------------
 &nbsp;    Model Likelihood   Discrimination 
              Ratio Test         Indexes     
--------- ------------------ ----------------
 **Obs**     **LR chi2**          **R2**     

  1000          7.507            0.007479    

**sigma**      **d.f.**         **R2 adj**   

  25.16           1              0.006484    

**d.f.**    **Pr(> chi2)**        **g**      

   998         0.006147           2.464      
---------------------------------------------

Table: Fitting linear model: cholesterol ~ age


---------------------------------
 Min    1Q    Median   3Q    Max 
----- ------ -------- ----- -----
-75.7 -17.29  0.1067  17.72 82.28
---------------------------------

Table: Residuals


--------------------------------------------------
    &nbsp;        Coef    S.E.     t     Pr(>|t|) 
--------------- -------- ------- ------ ----------
 **Intercept**  190.0202 4.10425 46.298  0.000000 

    **age**      0.2201  0.08027 2.742   0.006211 
--------------------------------------------------

Table: Coeficients
```

Rendering as:

current-pander-output

Adding the TeX formulas shouldn't be that hard, e.g. for the first table:

```
---------------------------------------------
 &nbsp;    Model Likelihood   Discrimination 
              Ratio Test         Indexes     
--------- ------------------ ----------------
   Obs         LR $\chi^2$         $R^2$     

  1000          7.507            0.007479    

 &nbsp;

  sigma          d.f.          $R^2_{adj}$   

  25.16           1              0.006484    

 &nbsp;

  d.f.      _Pr(> $\chi^2$)_        _g_      

   998         0.006147           2.464      
---------------------------------------------
   
```

Rendering as:

possible-pander-output

What do you think?

**Comment by harrelfe** [comment]
I think that with $ $ it's getting close but what you are really losing is the ability to have two fields in one LaTeX field such as the name of the index and its value separated by \hfill.  

Try your approach on an lrm fit with lots of missing values to get a stiffer test.  You won't get the dot chart showing the number of missing values across variables.  Note that the approach I was proposing would call latex and htlatex or latexml during executing of the R chunk and would be invisible to the user except for taking longer.

**Comment by daroczig** [comment]
We won't be able (and do not want) to reproduce the LaTeX or HTML representations of the models in markdown, as markdown is a lot simpler than those markup languages. But `pander` is an R package to turn R objects into markdown (not HTML and/or TeX), so we try to get the maximum out of the markdown features. I think the column/row-spanning is the top-most limitations compared to PDF/HTML :(

Anyway, thank you very much for your kind feedback, I hope we can further improve the markdown representations of the models.

**Comment by harrelfe** [comment]
The fact that users of RStudio who use Rmarkdown ultimately produce html means that you don't need to avoid producing html dynamically with R, and there are packages that make insertion of customized html into the Rmarkdown stream relatively easy.  If you want to stick with pure markdown, if you can specify left and right alignment of table cells you could simulate the LaTeX behavior by splitting certain columns into 2 columns with the first one left aligned and the 2nd one right aligned.  I assume that markdown will allow you to selectively include vertical lines separating some of the columns.

**Comment by daroczig** [comment]
We will stick with markdown, that's the goal of this package. BTW I believe in a life beyond RStudio and HTML :)

But unfortunately, there is no support for defining horizontal/vertical lines in a markdown table, nor for column spanning, so although we can split the labels and values (like "df" and "1") into two cells with left and right alignment, this way we cannot center the "Model Likelihood Ratio Test" above both. But I hope we can come up with a solution that makes everyone happy.

**Comment by harrelfe** [comment]
Thanks for the note.  I feel that we need a higher-level table markup language or command language.  The tables package in R is a pretty general model, and has a latex method (I also enhance this in the Hmisc package).  An html method for tables that is general would be nice but a new table markup language will probably benefit us all.

**Comment by Hjalt** [comment]
Lovely!

I noticed a minor oddity in that a column, `Type`, is produced using `pander(summary(model))`, but isn't so with the default print method.
```
require(rms)
require(pander)

set.seed(1)
n <- 1000
age <- rnorm(n, 50, 10)
surv <- sample(0:1, n, T)

dd <- datadist(age)
options(datadist = 'dd')    

fit <- lrm(surv ~ age)
```
Here, `summary(fit)` gives

```
             Effects              Response : surv 

 Factor      Low    High   Diff.  Effect  S.E.     Lower 0.95 Upper 0.95
 age         43.026 56.884 13.858 0.11651 0.085035 -0.050151  0.28318   
  Odds Ratio 43.026 56.884 13.858 1.12360       NA  0.951090  1.32730   
```

, while `pander(summary(fit))` gives

```
---------------------------------------------------------------------------------------
     &nbsp;       Low   High   Diff.   Effect   S.E.    Lower 0.95   Upper 0.95   Type 
---------------- ----- ------ ------- -------- ------- ------------ ------------ ------
    **age**      43.03 56.88   13.86   0.1165  0.08503   -0.05015      0.2832      1   

 **Odds Ratio**  43.03 56.88   13.86   1.124     NA       0.9511       1.327       2   
---------------------------------------------------------------------------------------
```

I assume that it's used to denote whether the row refers to the log odds or the odds ratio, but that it's pruned by the `print` function of the package.

**Comment by RomanTsegelskyi** [comment]
Nice catch. I fixed it in https://github.com/Rapporter/pander/commit/28b2fc26ff5b7fcebf93c2e552e32c6af6d859dc

**Pull request merged**